unison-runtime/src/Unison/Runtime/Serialize.hs

```diff
@@ -179,27 +174,41 @@ putFoldable putA as =
 putMap :: (a -> Builder) -> (b -> Builder) -> Map a b -> Builder
 putMap putA putB m = putMapping putA putB $ Map.toList m
 
-getList :: (MonadGet m) => m a -> m [a]
-getList a = getLength >>= (`replicateM` a)
+getList :: (PrimBase m) => Get m a -> Get m [a]
+getList ga = getVarInt >>= grab []
+  where
+    grab as (n :: Int)
+      | n <= 0 = evaluated $ reverse as
+      | otherwise = ga >>= \a -> grab (a : as) (n - 1)
```

Comment:
Do you know if it makes any difference if you just define this directly? Like just directly create a `Get m [a]` via a `ByteString -> Ix m -> m [a]`.

Also it seems like you should build up the list strictly. You don't want to build up n levels of thunks, only to have them get forced... or is strictness analysis taking care of this?

Another option is just allocate a mutable array once you know the length, then convert to a list at the end.

The other list / map decoders could be done similarly if any of the above is a performance boost.